clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -252,9 +269,19 @@ private void getNextBlock() throws IOException {
     }
 
     if (mTachyonStorageType.isStore()) {
-      mCurrentBlockOutStream =
-          mContext.getTachyonBlockStore().getOutStream(getNextBlockId(), mBlockSize, mHostname);
-      mShouldCacheCurrentBlock = true;
+      String hostname;
+      try {
+        hostname =
+            mHostname == null
```

Comment:
Could we remove hostname as an outstream option? Could it be replaced with a location policy for writing to a specified host? Right now it's confusing because users can specify both, but the location policy will be ignored.